test/e2e/upgrade/manifestdelete/manifest-delete.go

```diff
@@ -0,0 +1,192 @@
+package manifestdelete
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	restclient "k8s.io/client-go/rest"
+	apiregistrationsclientset "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+// UpgradeTest contains artifacts used during test.
+type UpgradeTest struct {
+	oc     *exutil.CLI
+	config *restclient.Config
+}
+
+type object struct {
+	kind      string
+	namespace string
+	name      string
+}
+
+type unstructured struct {
+	name      string
+	namespace string
+	group     string
+	version   string
+	resource  string
+}
+
+var (
+	// Update this array to contain objects which should have been deleted in this release. For example:
+	//	object{kind: "Namespace",
+	//		namespace: "",
+	//		name:      "openshift-foobar"},
+	//
+	deletes = []object{}
+
+	// Update this array to contain unstructured objects which should have been deleted in this release. For example:
```

Comment:
is there something that prevents someone from using the delete annotation and not adding it to this test?  I don't think it would occur to me to modify this test when adding the annotation to my manifest in my operator.

Reply:
OTA will most likely have to maintain